src/pudl_archiver/cli.py

```diff
@@ -73,21 +74,29 @@ def parse_main():
         action="store_true",
         help="Regenerate metadata from PUDL data source rather than existing archived metadata.",
     )
-    return parser.parse_args()
+    parser.add_argument(
+        "--depositor",
+        type=str,
+        default="zenodo",
+    )
+    return parser.parse_args(args)
 
 
-def main():
+async def archiver_entry(args=None):
     """Run desired archivers."""
     load_dotenv()
     logger.setLevel(logging.INFO)
     log_format = "%(asctime)s [%(levelname)8s] %(name)s:%(lineno)s %(message)s"
     coloredlogs.install(fmt=log_format, level=logging.INFO, logger=logger)
-    args = parse_main()
+    args = parse_main(args)
```

Comment:
Nit - it might be cleaner to not bother at all with arg parsing within this method, and just have it take various args as inputs. Then you get:

```python
def parse_main(): # no need to pass in an args object anymore, this always reads from sys.argv
    ...

async def archiver_entry(args:argparse.Namespace):
    ...
    datasets = args.datasets # just use the args object that was passed in

def main():
    args = parse_main()
    asyncio.run(archiver_entry(args))
```

And in test, where you use `archiver_entry(some_special_test_args)` that should all still work. This avoids passing the desired override args all the way into `parse_main()` just to have that pass things back out again.

Reply:
In this case I'm not sure how to parse the args from the test context before passing in to `archiver_entry`. I'm going to go ahead and merge for now since I'm leaving soon, but happy to change this later if there's something I'm missing.